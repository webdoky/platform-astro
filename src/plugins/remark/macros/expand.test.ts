import type { Root } from 'mdast';
import { describe, expect, test } from 'vitest';

import expandMacros from './expand.js';

describe('expandMacros', () => {
  test('expands macros', () => {
    const tree: Root = {
      type: 'root',
      children: [
        {
          type: 'text',
          value: `Докладно про стандартні вбудовані об'єкти: {{jsxref("Array")}}, {{jsxref("Boolean")}}, {{jsxref("Date")}}, {{jsxref("Error")}}, {{jsxref("Function")}}, {{jsxref("JSON")}}, {{jsxref("Math")}}, {{jsxref("Number")}}, {{jsxref("Object")}}, {{jsxref("RegExp")}}, {{jsxref("String")}}, {{jsxref("Map")}}, {{jsxref("Set")}}, {{jsxref("WeakMap")}}, {{jsxref("WeakSet")}}, та інші.`,
        },
      ],
    };
    expandMacros(tree);

    expect(tree).toMatchSnapshot();
  });
});
