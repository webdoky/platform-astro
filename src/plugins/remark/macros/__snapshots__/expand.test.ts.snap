// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`expandMacros > expands macros 1`] = `
{
  "children": [
    {
      "type": "text",
      "value": "Докладно про стандартні вбудовані об'єкти: ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;Array&quot;">{{jsxref(&quot;Array&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;Boolean&quot;">{{jsxref(&quot;Boolean&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;Date&quot;">{{jsxref(&quot;Date&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;Error&quot;">{{jsxref(&quot;Error&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;Function&quot;">{{jsxref(&quot;Function&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;JSON&quot;">{{jsxref(&quot;JSON&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;Math&quot;">{{jsxref(&quot;Math&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;Number&quot;">{{jsxref(&quot;Number&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;Object&quot;">{{jsxref(&quot;Object&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;RegExp&quot;">{{jsxref(&quot;RegExp&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;String&quot;">{{jsxref(&quot;String&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;Map&quot;">{{jsxref(&quot;Map&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;Set&quot;">{{jsxref(&quot;Set&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;WeakMap&quot;">{{jsxref(&quot;WeakMap&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", ",
    },
    {
      "type": "html",
      "value": "<span class="macro missing" data-macro-name="jsxref" data-parameters="&quot;WeakSet&quot;">{{jsxref(&quot;WeakSet&quot;)}}</span>",
    },
    {
      "type": "text",
      "value": ", та інші.",
    },
  ],
  "type": "root",
}
`;
